src/common/styles/selectors.css

```diff
@@ -4,7 +4,31 @@
  * In Thema the approach to CSS naming is, as much as possible, to
  * use custom selectors for the type of entities being styled and
  * their properties.
- */
+ *
+ * The top level selectors should target block level semantic elements, and avoid
+ * styling generic elements such as `div`s and `span`s. This limitation helps ensure that
+ * themes cause minimal conflicts with existing layouts and deeply nested components.
+ *
+*/
+@custom-selector :--pre pre, pre[class*="language-"];
```

Comment:
I'm a little uncomfortable with this use of custom selector for a non-semantic pre-formatted element. It feels a little ad-hoc. All the other uses of custom selectors below relate to an `itemtype`, an `itemprop` or containers for `itemprops` (e.g. `authors`). Looking below at the use of `:--pre`, it seem that to be consistent with that, we need a `@custom-selector :--CodeBlock` [itemtype='https://schema.stenci.la/CodeBlock']; and for Encoda to encode an itemtype for `CodeBlock` elements.

Generally, I think we should:

- decide which Schema node type we are _not_ going to add `itemtype` to in the HTML encoding (adding to `p`, `li`, `strong` etc is probably over the top)
- in Encoda add `itemtype` to HTML encodings of all other node types
- in Encoda, or here, autogenerate the `@custom-selector`s for the types and properties